File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int i=0,j=0;
        int n=nums.size();
        // while(i<n && nums[i]<0) i++;
        // while(j<n && nums[j]>=0) j++;
        vector<int> res;
        while(i<n && j<n){
            while(i<n && nums[i]<0) i++;
            res.push_back(nums[i++]);
            while(j<n && nums[j]>=0) j++;
            res.push_back(nums[j++]);
        }

        while(i<n && nums[i]>=0) res.push_back(nums[i++]);
        while(j<n && nums[j]<0) res.push_back(nums[j++]);
        return res;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0]*n
        pos,neg=0,1 
        for i in nums: 
            if i<0:
                res[neg]=i 
                neg+=2 
            else: 
                res[pos]=i 
                pos+=2 
        return res